rs/bitcoin/kyt/src/state.rs

```diff
@@ -45,10 +46,48 @@ pub enum FetchTxStatus {
 /// input address will be computed and filled in.
 #[derive(Clone, Debug)]
 pub struct FetchedTx {
-    pub tx: Transaction,
+    pub tx: TransactionInputOutput,
     pub input_addresses: Vec<Option<Address>>,
```

Comment:
nit: maybe also a bit late to the party here, but why not a `BTreeSet`?

Reply:
The index matters here too. We use the index to identify which input_address it is when filling in this array after an address is computed.